### ISSUE TITLE
Nightly 2019-12-29 causes `reached the type-length limit` error in previously working code

### ISSUE THREAD

**Issue by sdroege** [rust-lang/rust#67757]
`cargo bisect-rustc` considers 3a087ad3a924be12343bb035bf9b63ed81f650bf the culprit.

Can be reproduced on https://github.com/sdroege/async-tungstenite by running `cargo test`:

```
error: reached the type-length limit while instantiating `std::thread::LocalKey::<std::cel...ure::Future, ()}]>, ()}]>]], ()>`
    |
    = note: consider adding a `#![type_length_limit="1364215"]` attribute to your crate

error: aborting due to previous error

error: could not compile `async-tungstenite`.
warning: build failed, waiting for other jobs to finish...
error: reached the type-length limit while instantiating `async_task::utils::abort_on_pani..., async_std::task::task::Task>]>`
  --> /home/slomo/.cargo/registry/src/github.com-1ecc6299db9ec823/async-task-1.1.1/src/utils.rs:8:1
   |
8  | / pub(crate) fn abort_on_panic<T>(f: impl FnOnce() -> T) -> T {
9  | |     struct Bomb;
10 | |
11 | |     impl Drop for Bomb {
...  |
20 | |     t
21 | | }
   | |_^
   |
   = note: consider adding a `#![type_length_limit="1363798"]` attribute to your crate

error: aborting due to previous error

error: could not compile `async-tungstenite`.
warning: build failed, waiting for other jobs to finish...
error: reached the type-length limit while instantiating `std::thread::LocalKey::<std::cel...::Message>>, ()}]>, ()}]>]], ()>`
    |
    = note: consider adding a `#![type_length_limit="1367031"]` attribute to your crate

error: aborting due to previous error

error: could not compile `async-tungstenite`.
```

**Comment by jonas-schievink** [comment]
Presumably due to https://github.com/rust-lang/rust/pull/65244. cc @seanmonstar

**Comment by pnkfelix** [comment]
triage: P-high. We may well revert PR #65244; see ongoing discussion in issue #67706

**Comment by tmandry** [comment]
Fixed by #67768.
